### PR TITLE
DSPy Landing Page

### PR DESCRIPTION
**Pull request by erika-cardenas**
### Type of change:
- [x] **Documentation** updates (non-breaking change to fix/update documentation)

### How Has This Been Tested?
- [x] **Github action** – automated build completed without errors
- [x] **Local build** - the site works as expected when running `yarn start`

### CODE REVIEW

File: developers/weaviate/more-resources/dspy.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+1. [**Hands on Learning**](#hands-on-learning): Content framed to build your technical understanding with end-to-end tutorials. 
+
+2. [**Read and Listen**](#read-and-listen): Content designed to help develop your conceptual understanding of these technologies.
+
+
+### Hands on Learning 
+
+| Topic | Details |
+| --- | --- |
+| **Getting Started with RAG in DSPy** | Learn about the four components needed to build a DSPy program: Installation, settings, Datasets with `dspy.Example`, LLM Metrics, The DSPy programming model, and Optimization. [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/dspy/1.Getting-Started-with-RAG-in-DSPy.ipynb) and [Video](https://youtu.be/CEuUG4Umfxs?si=4Gp8gR9glmoMJNaU) |
```
Comment (daveatweaviate) [inline developers/weaviate/more-resources/dspy.md:36]: These should all be lower case:
Installation, settings, Datasets with `dspy.Example`, LLM Metrics, The DSPy programming model, and Optimization

File: developers/weaviate/more-resources/dspy.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+| **DSPy + Weaviate for the Next Generation of LLM Apps** | Build a 4-layer DSPy program for generating blog posts from queries. [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/dspy/2.Writing-Blog-Posts-with-DSPy.ipynb) and [Video](https://youtu.be/ickqCzFxWj0?si=AxCbD9tq2cbAH6bB)|
+| **RAG with Persona** | Build a compound AI system with DSPy, Cohere, and Weaviate, adding persona to language model outputs. [Notebook](https://github.com/weaviate/recipes/tree/main/integrations/dspy/fullstack-recipes/RAGwithPersona) and [Post](https://twitter.com/ecardenas300/status/1765444492348243976)|
```
Comment (daveatweaviate) [inline developers/weaviate/more-resources/dspy.md:38]: Is persona a product? This should probably be: 

adding Persona  (product)
or
adding a persona  (not a product)

File: developers/weaviate/more-resources/dspy.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+| **Adding Depth to RAG Programs** | Enhancing DSPy programs by integrating unique input-output examples and multiple LLMs. [Notebook](https://github.com/weaviate/recipes/blob/main/integrations/dspy/3.Adding-Depth-to-RAG-Programs.ipynb) and [Video](https://youtu.be/0c7Ksd6BG88?si=YUF2wm1ncUTkSuPQ) |
+| **Hurricane: Writing Blog Posts with Generative Feedback Loops** | Introduction to Hurricane, a web app for demonstrating Generative Feedback Loops with blog posts. [Notebook](https://github.com/weaviate-tutorials/Hurricane) and [Blog](/blog/hurricane-generative-feedback-loops) |
```
Comment (daveatweaviate) [inline developers/weaviate/more-resources/dspy.md:40]: This should be lower case:

generative feedback loops

**Review by daveatweaviate — APPROVED** [review]
@erika-cardenas there are a couple of typos, otherwise this LGTM

Let me know if you want to fix them, otherwise I can merge this in

File: developers/weaviate/more-resources/dspy.md (hunk 1 above)
Comment (erika-cardenas) [reply developers/weaviate/more-resources/dspy.md:36]: Agreed!

File: developers/weaviate/more-resources/dspy.md (hunk 2 above)
Comment (erika-cardenas) [reply developers/weaviate/more-resources/dspy.md:38]: Agreed! Switched to: 
"Build a compound AI system with DSPy, Cohere, and Weaviate, where you'll add a persona to the language model."

File: developers/weaviate/more-resources/dspy.md (hunk 3 above)
Comment (erika-cardenas) [reply developers/weaviate/more-resources/dspy.md:40]: Agreed!

**Comment by erika-cardenas** [comment]
Just updated the doc from your feedback!

**Comment by daveatweaviate** [comment]
Thanks @erika-cardenas  - I merged the PR

**Pull request merged**